### ISSUE TITLE
Mininum split area size

### ISSUE THREAD

**Issue by jitsmaster** [bertrandg/angular-split#4]
One feature always useful is to set minimum size for an area. Ideally in pixels.

Would you like me to work on this feature for you?

**Comment by bertrandg** [comment]
Yes sure, 
I first planned this feature directly but had some trouble so I differed it.
I don't have a lot of time to add it soon and I want to add tests too so go for it if you want! :+1:

**Comment by denys-kostenko** [comment]
Hello. Please provide opportunity to set property minPercent throught @Input parameter.

**Comment by jitsmaster** [comment]
Just start to get the time to work on this issue.

**Comment by jitsmaster** [comment]
Let me know if you are planning to work on it yourself. I haven't had the chance to start on it yet.

**Comment by bertrandg** [comment]
@jitsmaster Not planned currently but I will tell you there before starting! ;)

**Comment by adyakov** [comment]
@bertrandg I've found [minSizePixel] property in SplitArea directive. This what I need but I'm confused a little about this code
`if (newSizePixelA <= areaA.minPixel && newSizePixelB < areaB.minPixel) {`
in SplitComponent. I guess it should be
`if (newSizePixelA <= areaA.minPixel || newSizePixelB < areaB.minPixel) {`

**Comment by bertrandg** [comment]
Yes you are right but this feature isn't working currently.
This is why this property is not documented.

**Comment by adyakov** [comment]
@bertrandg it looks like it works properly if you make a fix from my code above

**Comment by bertrandg** [comment]
Cool if it works but I need to make some tests.
I'm gonna work on this feature in coming weeks.

**Comment by webmatrixxxl** [comment]
Hello! I also need this feature and/or maybe way to set fixed initial size go split area.

**Comment by synox** [comment]
The fix https://github.com/bertrandg/angular-split/issues/4#issuecomment-283282445 also works for me. But we would actually need a parameter "minPercent" for split-area.

**Comment by webmatrixxxl** [comment]
#4 (comment)  works, but minSizePixel is not set on init and that is a problem.

**Comment by kejtizuki** [comment]
@bertrandg are you planning to add a parameter "minPercent" to split-area?

**Comment by bertrandg** [comment]
hi, honestly not soon, got lot of work currently and not my first priority.

**Comment by MelissaVanDijk** [comment]
Hi Bertrand,

Even though the quick mouse move does not work yet, I added the fix of Adyakov to a pull request, to get the basic check on minSizePixel working. Can you merge this so we can use it in our project? We'll wait very patiently for the (less urgent) quick mouse move fix :-) 

Thanks in advance!

**Comment by dungahk** [comment]
I've submitted a PR yesterday fixing the problem, today I realized there was still a problem with my fix, if you set the min size to a size less than gutterSize/2, you would get bad results. Now I am udpating the #36  PR fixing this problem forcing the min size to be at least gutterSize/2.

I hope @bertrandg can test it and merge it soon.

**Comment by webmatrixxxl** [comment]
@dungahk [minSizePixel] in 0.2.0  is not working for me.

**Comment by dungahk** [comment]
@webmatrixxxl that's because the PR hasn't been merged yet, it's still the same.

**Comment by zwtokio** [comment]
you can configure, by going to this link :` \node_modules\angular-split\dist\split.component.js`

**Comment by mrkvw** [comment]
Any update on if the PR resolving this will be merged?

**Comment by stan-kondrat** [comment]
@bertrandg Thank you for your work, this is very useful component.  👍

Min size pixel is last missing feature for most use cases. 🙂

**Comment by rkopyrkin** [comment]
@bertrandg Hi Bertrand,
Do you plan to implement this feature? It will be very useful to have it. As far as I can see @dungahk already submitted PR for that, could you review and merge it?
Thanks!

**Comment by bertrandg** [comment]
hi,
Yes I plan to add min percentage size but I need time for it..

**Comment by jeevandark** [comment]
Hi - thanks for the valuable & excellent component @bertrandg ...
Me too would love to see that PR merged and published.
👍 
Thanks & cheers!

**Comment by TomashVlad** [comment]
Is there posibillity to implement this for 0.2.x version? This is only thing that is left that I need to use with this split? I am using Angular 2 on my project and this is component that is solving all my problems. I'll be so grateful if min size could be implemented. Thanks!!!

**Comment by filosof86** [comment]
Hi,

Thank you for your work @bertrandg and as it was said above it would be great if we could get the discussed feature or maybe some workaround for it if any.
Is there any news about it?

Thanks in advance

**Comment by chetanladdha** [comment]
I am eagerly looking out for this feature, Is there any workaround I can make in my code to make Min-size and Max-Size to work?

**Comment by chetanladdha** [comment]
Any plan on this feature to be merged?

**Comment by smiles2424** [comment]
There hasn't been any activity from the contributors in quite some time. I'm thinking we might need to fork. -.-

**Comment by smiles2424** [comment]
I have a new PR #111 for this work and I published a new package for those of us that need this immediately. https://www.npmjs.com/package/angular-split-minsize

**Comment by chetanladdha** [comment]
@smiles2424 : I am not able to find your npm package. Have you removed/renamed it?

**Comment by smiles2424** [comment]
sorry, I got a little ahead of myself. I had assumed if I merged the existing PR for minSize it would work fine but that is not the case unfortunately :/

**Comment by danito4** [comment]
Hello all, is there any updated for this please? Or maybe a workaround?

I have also tried to implement the following but it's still not working :(
https://stackoverflow.com/questions/48370025/how-to-set-minimum-and-maximum-margin-left

Thanks in advance!

**Comment by askri2pac** [comment]
hello, @bertrandg , have you merge the @dungahk  PR please ?

**Comment by guguji5** [comment]
Hi guys, I find this splitter is so cool and works well, but my PM's requirement is so wired. He wants to apply the max and min to every panel and when drag the first panel, give its width to the third panel.(for example there are three panel, default its width should be add to the second panel.)

So I refer to the splitter and write [a new one.(https://github.com/guguji5/angular-splitter)](https://github.com/guguji5/angular-splitter) I give up some unimportant points and add the mininum or maxinum to it.

Glad to discussion or suggestion.

**Comment by shirkedar** [comment]
Hi @bertrandg , Is there a time estimate for merging @dungahk fix?

**Comment by dungahk** [comment]
Truth is, I don't really know if the PR works anymore, it's been years since I opened the PR. Also, the 1.* version has been rewritten after the PR.

**Comment by bertrandg** [comment]
Check #131 for what's coming.
